am6[2|4]xx-sk-prepare: add note about the existing variants (#548)
foundriesio/docs: source/reference-manual/boards/am62xx-sk-prepare.rst

Building
--------

> When building the Cortex R5 u-boot, it build for all different variants:

Reviewer: ```suggestion
When building the Cortex R5 u-boot, it builds for all different variants:
```

* ``hs-fs`` (High Security/Field Securable) also known as `SK-AM62B`_: This is a SoC/board state before a customer has blown the keys in the device. i.e. the state at which HS device leaves TI factory. In this state, the device protects the ROM code, TI keys and certain security peripherals. In this state, device do not force authentication for booting, however DMSC is locked.

> * ``hs`` (High Security/Security Enforced) also known as `SK-AM62B`_: This is a SoC/board state after a customer has successfully blown the keys and set “customer Keys enable”. In HS-SE device all security features enabled. All secrets within the device are fully protected and all of the security goals are fully enforced. The device also enforces secure booting.

Reviewer: ```suggestion
* ``hs`` (High Security/Security Enforced) also known as `SK-AM62B`_: This is a SoC/board state after a customer has successfully blown the keys and set “customer keys enable”. In HS-SE device all security features are enabled. All secrets within the device are fully protected and all of the security goals are fully enforced. The device also enforces secure booting.
```

> * ``gp`` (General Purpose) variant also known as `SK-AM62`_: This is a SoC/board state where there is no devie protection and authentication is not enabled for booting the device.

Reviewer: ```suggestion
* ``gp`` (General Purpose) variant also known as `SK-AM62`_: This is a SoC/board state where there is no device protection and authentication is not enabled for booting the device.
```

> The default variant is the ``hs-fs`` and to boot an image on the others variants without pre-flash files manipulations
> on the target file-system we need to change the ``SYSFW_SUFFIX`` variable.

Reviewer: ```suggestion
The default variant is the ``hs-fs``. To boot an image on the others variants without pre-flash files manipulations
on the target file-system we need to change the ``SYSFW_SUFFIX`` variable.
```

> The following will change the default to ``gp`` and so the image produced will boot on that variant:

Reviewer: ```suggestion
The following changes the default to ``gp`` so the image produced boots on that variant:
```

.. code-block:: shell

>    $ cat meta-subscriber-overrides/conf/machine/include/lmp-factory-custom.inc | grep SYSFW_SUFFIX

Reviewer: An echo here would be better as we had customer issues with cat in the past, please check if the syntax is correct
```suggestion
   $ echo "SYSFW_SUFFIX:am62xx-evm-k3r5 = "gp"" >> meta-subscriber-overrides/conf/machine/include/lmp-factory-custom.inc
```

Preparation
-----------

The default variant of R5 u-boot is ``hs-fs`` (High Security/Field Securable) also known as SK-AM62B ``https://www.ti.com/tool/SK-AM62B``.
For building to the ``gp`` (General Purpose) variant also known as SK-AM62 ``https://www.ti.com/tool/SK-AM62``,
add ``SYSFW_SUFFIX:am62xx-evm-k3r5 = "gp"`` to ``meta-subscriber-overrides/conf/machine/include/lmp-factory-custom.inc``.

> The override ``am62xx-evm-k3r5`` is just for this machine. Using ``k3r5`` also works

Reviewer 2: I would remove 'Using ``k3r5`` also works or other k3r5 machines, such as ``am64xx-evm-k3r5``. just to avoid confusion, same for am64 -> am62.

Better to isolate the instructions for the target machine here.

Other than this, LGTM

Author: sometimes less information is better, keep it simple!

Preparation
-----------

The default variant of R5 u-boot is the ``hs-fs`` so for building to the ``gp`` variant
we need to change it on some global configuration file::

>       echo 'SYSFW_SUFFIX:am62xx-evm-k3r5 = "gp"' >> cong/local.conf

Reviewer 2: We don't recommend changing conf/local.conf as that is just for local development only.

For these changes we prefer changing the meta-subscriber-overrides layer as that is how an user would do in a factory.

Author: Thanks, I will adapt it to use something in meta-subscriber-overrides.

Author: fixed

Preparation
-----------

> The default variant of R5 u-boot is the ``hs-fs`` so for building to the ``gp`` variant

Reviewer 2: @quaresmajose can you add a quick note explaining shortly what HS / GP means? Even if just a link to the TI website, otherwise people won't necessarily know what to do here.

> The default variant of R5 u-boot is the ``hs-fs`` so for building to the ``gp`` variant
> we need to change it on some global configuration file, like::

Reviewer 3: ```suggestion
The default variant of R5 u-boot is ``hs-fs`` (High Security/Field Securable). For building to the ``gp`` (General Purpose) variant,
add ``SYSFW_SUFFIX:am62xx-evm-k3r5 = "gp"``to ``meta-subscriber-overrides/conf/machine/include/lmp-factory-custom.inc``.
```

      $ cat meta-subscriber-overrides/conf/machine/include/lmp-factory-custom.inc

>       SYSFW_SUFFIX:am62xx-evm-k3r5 = "gp"

Reviewer 2: @vanmaegima I know this cat process caused confusion in the past (user leading to think they had to replace the content instead), should we change to echo >> here instead?

Reviewer: An echo would be better for sure, or just mention "add the following line to lmp-factory-custom".

Reviewer 3: I would go with "add the following line...".

Author: fixed! thanks

> The override ``am62xx-evm-k3r5`` is just for this machine, the ``k3r5`` also works

Reviewer 3: ```suggestion
The override ``am62xx-evm-k3r5`` is just for this machine. Using ``k3r5`` also works
```

> and can be used for a broader range so it also works on other k3r5 like ``am64xx-evm-k3r5``.

Reviewer 3: ```suggestion
for other k3r5 machines, such as ``am64xx-evm-k3r5``.
```
